### PR TITLE
fix scala-js deps

### PR DESCRIPTION
**Pull request by sbrunk**
Essentially https://github.com/jwt-scala/jwt-scala/pull/460 without removing JS from the build and cherry-picked onto current master.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -156,6 +155,11 @@ val noPublishSettings = Seq(
   publishArtifact := false
 )
 
+lazy val commonJsSettings = Seq(
+  Test / fork := false,
+  Test / test := (()) // TODO
```
Comment (erwan) [inline build.sbt]: Why is there a TODO here?

Comment (sbrunk) [reply build.sbt]: With a bit of refactoring, we can re-enable some tests on JS. If we do, we need to remove this setting again though. I think that's why @massimosiani added the TODO.

Actually let me try to get the tests working that require the least effort.

Comment (sbrunk) [reply build.sbt]: `JwtClaimSpec` and `JwtBase64Spec` in `coreJS` are running and passing now (the regular and the test code is actually compiled to JavaScript and the tests run on Node.js but that's nicely abstracted away fortunately).

I think we might be able to enable more tests (i.e. in `circeJS`) but that needs more work, so I'd suggest to keep it out of this PR.

**Comment by erwan** [comment]
@massimosiani what's your take on this, OK to merge it?

**Comment by massimosiani** [comment]
Let's merge than.

**Review by sideeffffect — APPROVED** [review]
Let's merge this :+1: I'll update then https://github.com/jwt-scala/jwt-scala/pull/481 accordingly

**Pull request merged**